pulsar-common/pom.xml

```diff
@@ -74,6 +74,11 @@
       <artifactId>netty-all</artifactId>
     </dependency>
 
+    <dependency>
```

Comment:
@merlimat you need to add this dependency to be included in  all the other shaded modules that depend on pulsar-common or the following exception will happen:

```
java.lang.NoSuchMethodError: org.apache.bookkeeper.common.allocator.ByteBufAllocatorBuilder.pooledAllocator(Lorg/apache/pulsar/shade/io/netty/buffer/ByteBufAllocator;)Lorg/apache/bookkeeper/common/allocator/ByteBufAllocatorBuilder;
   at org.apache.pulsar.common.allocator.PulsarByteBufAllocator.<clinit>(PulsarByteBufAllocator.java:66)
   at org.apache.pulsar.client.impl.ConnectionPool.<init>(ConnectionPool.java:77)
   at org.apache.pulsar.client.impl.ConnectionPool.<init>(ConnectionPool.java:62)
   at org.apache.pulsar.client.impl.PulsarClientImpl.<init>(PulsarClientImpl.java:121)
   at org.apache.pulsar.client.impl.PulsarClientImpl.<init>(PulsarClientImpl.java:117)
   at org.apache.pulsar.client.impl.ClientBuilderImpl.build(ClientBuilderImpl.java:61)
```

Also this bookkeeper dependency needs to be shaded as well to prevent clashes with user dependencies